src/main/java/org/opensearch/security/ssl/SecureSSLSettings.java

```diff
@@ -16,115 +16,77 @@
-                .map(SecureString::toString)
-                .orElse(defaultValue);
-        }
-    }
+    // http settings
+    public final static SecurableLegacySetting SECURITY_SSL_HTTP_PEMKEY_PASSWORD = new SecurableLegacySetting(
+        HTTP_PREFIX + ".pemkey_password"
+    );
+    public final static SecurableLegacySetting SECURITY_SSL_HTTP_KEYSTORE_PASSWORD = new SecurableLegacySetting(
+        HTTP_PREFIX + ".keystore_password"
+    );
+    public final static SecurableLegacySetting SECURITY_SSL_HTTP_KEYSTORE_KEYPASSWORD = new SecurableLegacySetting(
+        HTTP_PREFIX + ".keystore_keypassword"
+    );
+    public final static SecurableLegacySetting SECURITY_SSL_HTTP_TRUSTSTORE_PASSWORD = new SecurableLegacySetting(
+        HTTP_PREFIX + ".truststore_password",
+        DEFAULT_STORE_PASSWORD
+    );
+
+    // transport settings
+    public final static SecurableLegacySetting SECURITY_SSL_TRANSPORT_PEMKEY_PASSWORD = new SecurableLegacySetting(
+        TRANSPORT_PREFIX + ".pemkey_password"
+    );
+    public final static SecurableLegacySetting SECURITY_SSL_TRANSPORT_SERVER_PEMKEY_PASSWORD = new SecurableLegacySetting(
+        TRANSPORT_PREFIX + ".server.pemkey_password"
+    );
+    public final static SecurableLegacySetting SECURITY_SSL_TRANSPORT_CLIENT_PEMKEY_PASSWORD = new SecurableLegacySetting(
+        TRANSPORT_PREFIX + ".client.pemkey_password"
+    );
+    public final static SecurableLegacySetting SECURITY_SSL_TRANSPORT_KEYSTORE_PASSWORD = new SecurableLegacySetting(
+        TRANSPORT_PREFIX + ".keystore_password"
+    );
+    public final static SecurableLegacySetting SECURITY_SSL_TRANSPORT_KEYSTORE_KEYPASSWORD = new SecurableLegacySetting(
+        TRANSPORT_PREFIX + ".keystore_keypassword"
+    );
+    public final static SecurableLegacySetting SECURITY_SSL_TRANSPORT_SERVER_KEYSTORE_KEYPASSWORD = new SecurableLegacySetting(
+        TRANSPORT_PREFIX + ".server.keystore_keypassword"
+    );
+    public final static SecurableLegacySetting SECURITY_SSL_TRANSPORT_CLIENT_KEYSTORE_KEYPASSWORD = new SecurableLegacySetting(
+        TRANSPORT_PREFIX + ".client.keystore_keypassword"
+    );
+    public final static SecurableLegacySetting SECURITY_SSL_TRANSPORT_TRUSTSTORE_PASSWORD = new SecurableLegacySetting(
+        TRANSPORT_PREFIX + ".truststore_password",
+        DEFAULT_STORE_PASSWORD
+    );
 
     private SecureSSLSettings() {}
 
     public static List<Setting<?>> getSecureSettings() {
-        return Arrays.stream(SSLSetting.values())
+        return Arrays.stream(SecureSSLSettings.class.getDeclaredFields())
+            .filter(field -> SecurableLegacySetting.class.isAssignableFrom(field.getType()))
+            .map(field -> {
+                try {
+                    return (SecurableLegacySetting) field.get(null);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Unable to access field: " + field.getName(), e);
```

Comment:
It took me a while to understand this until I figured out that you are using reflection to retrieve static members (as a replacement for enum.values()). 

As this is solution is maybe a bit non-obvious, I would recommend to add a comment somewhere to explain the approach.

Another question: With the SecurityManager, this would needed to be wrapped in a doPrivileged() block in order to be able to do reflection. With the new security manager replacement, did this need go away?